src/core/lis-histogram-element.ts

```diff
@@ -0,0 +1,276 @@
+
+  private resize(entries: ResizeObserverEntry[]) {
+    entries.forEach((entry: ResizeObserverEntry) => {
+      if (
+        entry.target == this._histogramContainerRef.value &&
+        entry.contentBoxSize
+      ) {
+        this.requestUpdate();
+      }
+    });
+  }
+
+  private histogramContainerReady() {
+    if (this._histogramContainerRef.value) {
+      this.resizeObserverController.observe(this._histogramContainerRef.value);
+    }
+  }
+
+  override render() {
+    this.renderHistogram(this._data);
+    return html`<div
+      ${ref(this._histogramContainerRef)}
+      ${ref(this.histogramContainerReady)}
+    ></div>`;
+  }
+  override createRenderRoot() {
+    return this;
+  }
+
+  renderHistogram(theHistogram: HistogramDataModel[]) {
+    if (!this._histogramContainerRef.value) return;
+    this._histogramContainerRef.value.innerHTML = '';
+    const maxLabelLength = d3.max(theHistogram, (d) => d.name.length) as number;
+    if (!maxLabelLength) return;
+    const padding = 9 * maxLabelLength; // padding around the SVG
+    const width = this._width - 2 * padding; // adjust width
+    const height = this._height - 2 * padding; // adjust height
+
+    const svgContainer = d3
+      .select(this._histogramContainerRef.value)
+      .append('svg')
+      .attr('width', width + 2 * padding)
+      .attr('height', height + 2 * padding)
+      .append('g')
+      .attr('transform', `translate(${padding}, ${padding})`);
+
+    const x = d3
+      .scaleBand()
+      .domain(theHistogram.map((d) => d.name))
+      .range([0, width])
+      .padding(0.1);
+
+    const y = d3
+      .scaleLinear()
+      .domain([0, d3.max(theHistogram, (d) => d.count) as number])
+      .range(this.orientation === 'vertical' ? [0, height] : [height, 0]);
+
+    // Create the bars
+    svgContainer
+      .selectAll('rect')
+      .data(theHistogram)
+      .enter()
+      .append('rect')
+      .attr('x', (d) =>
+        this.orientation === 'vertical' ? 0 : (x(d.name) as number),
+      )
+      .attr('y', (d) =>
+        this.orientation === 'vertical' ? (x(d.name) as number) : y(d.count),
+      )
+      .attr('width', (d) =>
+        this.orientation === 'vertical' ? y(d.count) : x.bandwidth(),
+      )
+      .attr('height', (d) =>
+        this.orientation === 'vertical' ? x.bandwidth() : height - y(d.count),
+      )
+      .attr('fill', 'steelblue')
+      .append('title') // append a title element to each rectangle
+      .text((d) => `${this._xlabel}: ${d.name}, ${this._ylabel}: ${d.count}`); // set the text of the title
+
+    // Add the x-axis label
+    svgContainer
+      .append('text')
+      .attr(
+        'transform',
+        'translate(' + width / 2 + ' ,' + (height + padding - 5) + ')',
+      )
+      .style('text-anchor', 'middle')
+      .style('fill', 'black')
+      .text(this.orientation === 'vertical' ? this._ylabel : this._xlabel);
+
+    // Add the y-axis label
+    svgContainer
```

Comment:
The edge of the yAxis label is being trimmed in the example. Can this be adjusted or the space allocated dynamically?